packages/sdk/src/publish/MessageFactory.ts

```diff
@@ -98,6 +100,9 @@ export class MessageFactory {
         const messageId = new MessageID(this.streamId, partition, msgRef.timestamp, msgRef.sequenceNumber, publisherId, msgChainId)
 
         const encryptionType = (await this.streamRegistry.hasPublicSubscribePermission(this.streamId)) ? EncryptionType.NONE : EncryptionType.AES
+        if (!isCompliantEncryptionType(encryptionType, this.config)) {
+            throw new Error(`Publishing to stream ${this.streamId} was prevented because configuration requires encryption!`)
```

Comment:
This could throw `StreamrClientError`.

(Actually all cases where we throw `new Error(`) could be converted to use `StreamrClientError`. Maybe most important for these kinds of user-facing errors as the error code helps the communication.)